ansible_runner/config/_base.py

```diff
@@ -516,6 +535,27 @@ def wrap_args_for_containerization(self, args, execution_mode, cmdline_args):
         logger.debug(f"container engine invocation: {' '.join(new_args)}")
         return new_args
 
+    def _generate_container_auth_file(self, host, username, password):
+        token = "{}:{}".format(username, password)
+        encoded_container_auth_data = {'auths': {host: {'auth': b64encode(token.encode('UTF-8')).decode('UTF-8')}}}
+        # Create a new temp file with container auth data
+        path = tempfile.mkdtemp(prefix='ansible_runner_registry_%s_' % self.ident)
+
+        @atexit.register
+        def cleanup_ansible_runner_registry():
+            shutil.rmtree(path)
```

Comment:
Since my PR has merged and you rebased, you could replace this with `register_for_cleanup(path)`, which is a pattern in a few other places now. But you don't have to.